### PR TITLE
Adding Agent v5 tests

### PR DESCRIPTION
**Pull request by ClaudioESSilva**
So we have a working version, but I still have questions about the best way of doing a couple of things.
But I think we can go through them on a call 😄 

# A New PR

THANK YOU - We love to get PR's and really appreciate your time and help to improve this module

## Accepting a PR

Before we accept the PR - please confirm that you have run the tests locally to avoid our automated build and release process failing. You can see how to do that in our wiki

https://github.com/sqlcollaborative/dbachecks/wiki 

## Please confirm you have 0 failing Pester Tests

[] There are 0 failing Pester tests

## Changes this PR brings

Please add below the changes that this PR covers. It doesnt need an essay just the bullet points :-)

### CODE REVIEW

**Review by jpomfret — APPROVED** [review]

File: source/checks/Agentv5.Tests.ps1 [hunk 1]
```diff
@@ -66,374 +71,176 @@ Describe "SQL Agent Account" -Tag AgentServiceAccount, ServiceAccount -ForEach $
...
+        if ($AgentAlertNotification) {
+            It "MessageID <_.AgentMessageID> Alert should have a notification on <_.InstanceName>" -Skip:$skipAgentAlerts -ForEach ($PSItem.AgentAlerts.MessageIDs) {
+                $PSItem.HasNotification -in 1, 2, 3, 4, 5, 6, 7 | Should -Be $true -Because "Should notify by Agent notifications"
+            }
+        }
+    }
+}
+
+Describe "Job History Configuration" -Tag JobHistory, Agent -ForEach $InstancesToTest {
+    $skipAgetJobHistory = ($__dbcconfig | Where-Object { $_.Name -eq 'skip.agent.JobHistory' }).Value
```
Comment (jpomfret) [inline source/checks/Agentv5.Tests.ps1:231]:
```suggestion
    $skipAgentJobHistory = ($__dbcconfig | Where-Object { $_.Name -eq 'skip.agent.JobHistory' }).Value
```

File: source/checks/Agentv5.Tests.ps1 [hunk 2]
```diff
@@ -66,374 +71,176 @@ Describe "SQL Agent Account" -Tag AgentServiceAccount, ServiceAccount -ForEach $
...
+    [int]$minimumJobHistoryRows = ($__dbcconfig | Where-Object { $_.Name -eq 'agent.history.maximumhistoryrows' }).Value
+
+    if ($minimumJobHistoryRows -eq -1) {
+        It "The maximum job history configuration should be set to disabled on <_.InstanceName>" -Skip:$skipAgetJobHistory -ForEach ($PSItem.JobHistory) {
```
Comment (jpomfret) [inline source/checks/Agentv5.Tests.ps1:235]:
```suggestion
        It "The maximum job history configuration should be set to disabled on <_.InstanceName>" -Skip:$skipAgentJobHistory -ForEach ($PSItem.JobHistory) {
```

File: source/checks/Agentv5.Tests.ps1 [hunk 3]
```diff
@@ -66,374 +71,176 @@ Describe "SQL Agent Account" -Tag AgentServiceAccount, ServiceAccount -ForEach $
...
+            $PSItem.CurrentMaximumHistoryRows | Should -Be $PSItem.ExpectedMaximumHistoryRows -Because "Maximum job history configuration should be disabled"
+        }
+    } else {
+        It "The maximum job history number of rows configuration should be greater or equal to <_.ExpectedMaximumHistoryRows> on <_.InstanceName>" -Skip:$skipAgetJobHistory -ForEach ($PSItem.JobHistory) {
```
Comment (jpomfret) [inline source/checks/Agentv5.Tests.ps1:239]:
```suggestion
        It "The maximum job history number of rows configuration should be greater or equal to <_.ExpectedMaximumHistoryRows> on <_.InstanceName>" -Skip:$skipAgentJobHistory -ForEach ($PSItem.JobHistory) {
```

File: source/checks/Agentv5.Tests.ps1 [hunk 4]
```diff
@@ -66,374 +71,176 @@ Describe "SQL Agent Account" -Tag AgentServiceAccount, ServiceAccount -ForEach $
...
+            $PSItem.CurrentMaximumHistoryRows | Should -BeGreaterOrEqual $PSItem.ExpectedMaximumHistoryRows -Because "We expect the maximum job history row configuration to be greater than the configured setting <_.ExpectedMaximumHistoryRows>"
+        }
+        It "The maximum job history rows per job configuration should be greater or equal to <_.ExpectedMaximumJobHistoryRows> on <_.InstanceName>" -Skip:$skipAgetJobHistory -ForEach ($PSItem.JobHistory) {
```
Comment (jpomfret) [inline source/checks/Agentv5.Tests.ps1:242]:
```suggestion
        It "The maximum job history rows per job configuration should be greater or equal to <_.ExpectedMaximumJobHistoryRows> on <_.InstanceName>" -Skip:$skipAgentJobHistory -ForEach ($PSItem.JobHistory) {
```

File: source/checks/Agentv5.Tests.ps1 (hunk 1 above)
Comment (ClaudioESSilva) [reply source/checks/Agentv5.Tests.ps1:231]: Thanks! Updated

File: source/checks/Agentv5.Tests.ps1 (hunk 2 above)
Comment (ClaudioESSilva) [reply source/checks/Agentv5.Tests.ps1:235]: Thanks! Updated

File: source/checks/Agentv5.Tests.ps1 (hunk 3 above)
Comment (ClaudioESSilva) [reply source/checks/Agentv5.Tests.ps1:239]: Thanks! Updated

**Pull request merged**